### ISSUE TITLE
Ruffle Error on https://e621.net/posts/2529242?q=flash

### ISSUE THREAD

**Issue by bootmii** [ruffle-rs/ruffle#2201]
# Error Info
Error name: CompileError
Error message: WebAssembly.instantiateStreaming(): Wasm code generation disallowed by embedder
Error stack:
```
CompileError: WebAssembly.instantiateStreaming(): Wasm code generation disallowed by embedder
    at <anonymous>:3:66785
    at Array.forEach (<anonymous>)
    at Object.vi.f.wasm (<anonymous>:3:66393)
    at <anonymous>:3:34737
    at Array.reduce (<anonymous>)
    at Function.vi.e (<anonymous>:3:34714)
    at i (<anonymous>:3:6432)
    at i.next (<anonymous>)
    at <anonymous>:3:6732
    at new Promise (<anonymous>)
```

# Player Info
Allows script access: false

# Page Info
Page URL: https://e621.net/posts/2529242?q=flash

# Browser Info
Useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36
OS: Win32

# Ruffle Info
Version: 0.1.0
Name: nightly 2021-01-01
Channel: nightly
Built: 2021-01-01T00:47:35.621Z
Commit: 8cdf04384c79c60e066c945566128b7e7ee2620b

**Comment by Kippykip** [comment]
ew

**Comment by SamMorrowDrums** [comment]
I think this issue should also be closed, and something added to a FAQ. We can't govern individual sites CSPs so if they can't run the plugin as it is blocked to prevent XSS then we just have to accept the block. The workaround is that the plugin won't work - but the content host can be contacted and asked to either embed Ruffle, or change their CSP. Otherwise :-/

As I mentioned on #2278 

Unfortunately for this is a security feature of browers, and presumably the content security policy of this website does now allow wasm to be run. The website owner would need to

https://bugs.chromium.org/p/chromium/issues/detail?id=948834
WebAssembly/content-security-policy#7

This is the sites CSP:

Content-Security-Policy: default-src 'self' data: ; script-src 'self' ; style-src 'self' 'unsafe-inline' data: ; frame-ancestors 'none' ; base-uri 'none' ;

Right now unsafe-eval needs to be allowed in order to use plugin.

**Comment by SamMorrowDrums** [comment]
Hmm, it's worth mentioning we might be able to make this work by adding `wasm-eval` to the content security manifest of the plugin itself, so worth trying before giving up.

**Comment by SamMorrowDrums** [comment]
https://chrome.google.com/webstore/detail/content-security-policy-o/lhieoncdgamiiogcllfmboilhgoknmpi?hl=en Also this could enable manual override in the meantime.

**Comment by Herschel** [comment]
I did give a brief try of using `wasm-eval` in the extension to make this work, but it didn't seem to help. I'll have to investigate a little more. We should display a better error message perhaps to avoid the issue spam.

**Comment by kotenok2000** [comment]
You can change security.csp.enable to false in about:config in firefox developer edition.
You can also change xpinstall.signatures.required to false to be able install ruffle permanently.

**Comment by rufflue** [comment]
i was able to access #3355 content with ruffle extension version 2021-02-21
![hunter ruffle](https://user-images.githubusercontent.com/64484245/108666530-e7f14100-749c-11eb-970f-2e24b38cdb86.png)

**Comment by Tynach** [comment]
> I did give a brief try of using `wasm-eval` in the extension to make this work, but it didn't seem to help. I'll have to investigate a little more. We should display a better error message perhaps to avoid the issue spam.

@Herschel, it seems you wrote this before Mozilla fixed [this bug][bug], which might have been what you were encountering at this time. Could you try this fix again and see if it now works?

[bug]: https://bugzilla.mozilla.org/show_bug.cgi?id=1770468

**Comment by AmyRoxwell** [comment]
> I think this issue should also be closed, and something added to a FAQ. We can't govern individual sites CSPs so if they can't run the plugin as it is blocked to prevent XSS then we just have to accept the block. The workaround is that the plugin won't work - but the content host can be contacted and asked to either embed Ruffle, or change their CSP. Otherwise :-/
> 
> As I mentioned on #2278
> 
> Unfortunately for this is a security feature of browers, and presumably the content security policy of this website does now allow wasm to be run. The website owner would need to
> 
> https://bugs.chromium.org/p/chromium/issues/detail?id=948834 [WebAssembly/content-security-policy#7](https://github.com/WebAssembly/content-security-policy/issues/7)
> 
> This is the sites CSP:
> 
> Content-Security-Policy: default-src 'self' data: ; script-src 'self' ; style-src 'self' 'unsafe-inline' data: ; frame-ancestors 'none' ; base-uri 'none' ;
> 
> Right now unsafe-eval needs to be allowed in order to use plugin.

You are able to bypass this easily on e621 or e926, just click this button here
![image](https://github.com/ruffle-rs/ruffle/assets/89411895/ba91f783-835f-45b2-a191-8f7a5f67cfe2)
it will then launch a different page were the game or flash animation will run fine, obviously there are some bugs there due to ActionScript3 and 2 still not having implemented some stuff but most of them will work

An example of this actually working is this animation, if you click on "Save this flash" it will run the animation without any issues.
https://e621.net/posts/189261?q=flash+order%3Ascore+rating%3Asafe

**Comment by danielhjacobs** [comment]
Fixed (as well as possible) by #17152. When using the extension, the e261 pages and pages with similar issues will now ask you to open the the content in a new tab to sidestep the issue. This particular content seems to have been removed, however.
